### ISSUE TITLE
[CRASH] AIR 31.1.1.217 Starlings Texture.fromColor() crashes app, which is tied to Context3D.clear()

### ISSUE THREAD

**Issue by teotigraphix** [Gamua/Adobe-Runtime-Support#476]
## Problem Description

I updated to **AIR 31.1.1.217** Windows 10 today and recompiled my app on Windows(Android app). Once I ran the app, instant crash from the low level, no crash in the debugger, just dead app, player exits.

After further investigation I narrowed the crash down to `Context3D.clear()`.

## Steps to Reproduce

This is a total regression issues and has to affect more devs than just me.

## Known Workarounds

Remove calls from `Texture.fromColor()` which in turn call `Context3D.clear()`.

See;

https://forum.starling-framework.org/d/22714-air-3111217-texturefromcolor-crashes-app

**Comment by ajwfrost** [comment]
Thanks - just to confirm, this is when running the app in Windows via ADL? Strange that this one didn't get picked up in QA, let me double-check...

**Comment by teotigraphix** [comment]
> Thanks - just to confirm, this is when running the app in Windows via ADL? 

Check, through FB debugger.

**Comment by teotigraphix** [comment]
@ajwfrost Just to make it clear, there was absolutely nothing changed in my code or environment during recompile, this was same code, same assets, same PC, same dev, same gps coords but was not the same runtime experience with 217.  :)

**Comment by pol2095** [comment]
same issue using ADL on Windows

**Comment by ajwfrost** [comment]
Also checking ... are you doing render-to-texture at all in your applications? My suspicions are that this is related to a fix we did on stage3d rendering on Windows, but a simple call to Context3D.clear() is working fine, so it may be related to whatever it's actually clearing...

**Comment by ajwfrost** [comment]
... although I just checked the changes made there, and can't see how they would cause a crash. We'll try to get a test case to reproduce this here...

**Comment by teotigraphix** [comment]
> Also checking ... are you doing render-to-texture at all in your applications? 

Nope, my app that I am dealing with uses Feathers, so it was right at startup when the theme is loaded. So imagine the crash is before the native extensions are even loaded, it's literally doing nothing other than getting a blank color texture to use as a skin with Texture.fromColor().

I normally use `Quad` but I had 3 calls to `Texture.fromColor()` in startup theming, removed and replaced with `Quad`, everything is running fine in the debugger again.

**Comment by ajwfrost** [comment]
Okay thanks for all the info .. this is something that was introduced in the previous release (33.1.1.217) due to the fix for GPU memory leak: there was one point where we were releasing a COM object pointer before it was re-assigned with a new value .. however in some code paths, it wasn't actually a new value, but was the same object. So the reference counting went wrong and the code later crashed within D3D11 when we next used that pointer.

I can get a patched DLL uploaded if anyone needs this urgently?

**Comment by teotigraphix** [comment]
> I can get a patched DLL uploaded if anyone needs this urgently?

I'm good for now, since it's just the debugger and I could change the calls to Quad.

Thanks for checking into this so fast!
